### PR TITLE
don't filter False params to interfaces_attributes

### LINKED ISSUE #1148: Unable to create a host with unmanaged interface(s)
##### SUMMARY

It is impossible to create bare metal hosts with an unmanaged interface. 

From the code in plugins/modules/host.py line 480 to 483, there is:
```python
if 'interfaces_attributes' in module.foreman_params:
    filtered = [nic for nic in ({k: v for k, v in obj.items() if v} for obj in module.foreman_params['interfaces_attributes']) if nic]
    module.foreman_params['interfaces_attributes'] = filtered
```
This code above implies that any 'interfaces_attributes' parameter evaluated to false is removed from the API Call. But the default value of a parameter is not always "False". 

For example, it is possible to create and provision a bare metal host with multiple interfaces. The "managed" interface parameter default value is True. This lead to be unable to create bare metal hosts with an unmanaged interface. 

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.9.13
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/max/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/max/deploy/venvs_python/ansible_epdb_python_2/lib/python2.7/site-packages/ansible
  executable location = /home/max/deploy/venvs_python/ansible_epdb_python_2/bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### COLLECTION VERSION

```
redhat.satellite:1.5.1
```
Repo link: https://github.com/RedHatSatellite/satellite-ansible-collection
This was not tested with the community foreman collection version.

##### KATELLO/FOREMAN VERSION

```
tfm-rubygem-katello-3.16.0.13-1.el7sat.noarch
foreman-2.1.2.21-1.el7sat.noarch
```

##### STEPS TO REPRODUCE

```yaml
---
- hosts: localhost
  gather_facts: False
  tasks:

    - include_vars: input_variables/vault_secrets.yml

    - name: "Create Satellite host"
      theforeman.foreman.host:
        name: "example-host.example.com"
        architecture: "x86_64"
        build: True
        comment: "Test collection"
        domain: "example.com"
        interfaces_attributes:
          - subnet: "Example subnet provisioning"
            type: interface
            domain: "example.com"
            mac: 00:30:64:00:00:00
            managed: True
            primary: True
            provision: True
          - subnet: "Example subnet 2"
            type: interface
            domain: "example.com"
            identifier: eth1
            managed: False
            primary: False
            provision: False
            virtual: False
        location: "example_location"
        operatingsystem: "CentOS 7 Latest"
        organization: "example_organization"
        root_pass: "example_password"
        provision_method: build
        ptable: "Kickstart default"
        pxe_loader: "PXELinux BIOS"
        medium: "CentOS 7 mirror"
        server_url: "https://satellite-host"
        username: "example_user"
        password: "{{ satellite_user_password }}"
        validate_certs: False
        state: present
```
From the example above you can see the definition of an interface ("Example subnet 2") with the managed boolean set to False. But as this parameter is removed from the API call and the default value is True, this leads to errors.

##### EXPECTED RESULTS

```
PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************************

TASK [include_vars] **********************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Create Satellite host] *************************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```
##### ACTUAL RESULTS

```
PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************************

TASK [include_vars] **********************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Create Satellite host] *************************************************************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "error": {"errors": {"interfaces.mac": ["can't be blank"]}, "full_messages": ["Mac can't be blank"], "id": null}, "msg": "Error while performing create on hosts: 422 Client Error: Unprocessable Entity for url: https://satellite-host.example.com/api/hosts"}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

### PR DESCRIPTION
**Pull request by evgeni**
the filtering code is not required at all, as we have proper "is not
None" filtering as part of the foreman_helper already

Fixes: #1148

### CODE REVIEW

File: tests/test_playbooks/fixtures/host_interface_attributes-6.yml [hunk 1]
```diff
@@ -366,7 +366,7 @@ interactions:
       message: OK
 - request:
     body: '{"interface": {"mac": "aa:bb:cc:dd:ee:ff", "type": "interface", "identifier":
-      "eth0", "managed": true}}'
+      "eth0", "managed": true, "primary": false, "provision": false}}'
```
Comment (evgeni) [inline tests/test_playbooks/fixtures/host_interface_attributes-6.yml:369]: here is the real change in action :)

File: tests/test_playbooks/fixtures/host_interface_attributes-6.yml [hunk 2]
```diff
@@ -426,7 +426,7 @@ interactions:
       message: Created
 - request:
     body: '{"interface": {"mac": "aa:bb:cc:dd:ee:ee", "type": "interface", "identifier":
-      "eth1", "managed": true}}'
+      "eth1", "managed": true, "primary": false, "provision": false}}'
```
Comment (evgeni) [inline tests/test_playbooks/fixtures/host_interface_attributes-6.yml:429]: and here

File: plugins/modules/host.py [hunk 3]
```diff
@@ -477,10 +477,6 @@ def main():
         elif 'owner_group' in module.foreman_params:
             module.foreman_params['owner_type'] = 'Usergroup'
 
-        if 'interfaces_attributes' in module.foreman_params:
-            filtered = [nic for nic in ({k: v for k, v in obj.items() if v} for obj in module.foreman_params['interfaces_attributes']) if nic]
-            module.foreman_params['interfaces_attributes'] = filtered
```
Comment (mdellweg) [inline plugins/modules/host.py:482]: I should have been `if ... is None` anyway, right?

**Review by mdellweg — APPROVED** [review]

File: plugins/modules/host.py (hunk 3 above)
Comment (evgeni) [reply plugins/modules/host.py:482]: Correct, that's the actual bug here.

**Pull request merged**